src/back/msl/writer.rs

```diff
@@ -608,8 +608,17 @@ impl<W: Write> Writer<W> {
                 };
                 write!(self.out, "{}", coco)?;
             }
-            crate::Expression::Splat { size: _, value } => {
+            crate::Expression::Splat { size, value } => {
+                let scalar_kind = match *context.resolve_type(value) {
+                    crate::TypeInner::Scalar { kind, .. } => kind,
+                    _ => unreachable!()
```

Comment:
I'm not sure if we want to handle the case where `value` _isn't_ a scalar.

Reply:
let's return `Error::Validation` instead of panicking here